src/scaffoldmaker/utils/tubenetworkmesh.py

```diff
@@ -87,21 +176,34 @@ def isShowTrimSurfaces(self):
 
 class TubeNetworkMeshSegment(NetworkMeshSegment):
 
-    def __init__(self, networkSegment, pathParametersList, elementsCountAround, elementsCountThroughWall):
+    def __init__(self, networkSegment, pathParametersList, elementsCountAround, elementsCountThroughWall,
+                 isCore=False, elementsCountAcrossMajor: int = 4, elementsCountAcrossMinor: int = 4,
+                 elementsCountTransition: int = 1):
         """
         :param networkSegment: NetworkSegment this is built from.
         :param pathParametersList: [pathParameters] if 2-D or [outerPathParameters, innerPathParameters] if 3-D
         :param elementsCountAround: Number of elements around this segment.
         :param elementsCountThroughWall: Number of elements between inner and outer tube if 3-D, 1 if 2-D.
+        :param isCore: True for generating a solid core inside the tube, False for regular tube network.
+        :param elementsCountAcrossMajor: Number of elements across major axis of an ellipse.
+        :param elementsCountAcrossMinor: Number of elements across minor axis of an ellipse.
+        :param elementsCountTranstion: Number of elements across transition zone between core box elements and
+        rim elements.
         """
         super(TubeNetworkMeshSegment, self).__init__(networkSegment, pathParametersList)
+        self._isCore = isCore
         self._elementsCountAround = elementsCountAround
+        self._elementsCountAcrossMajor = elementsCountAcrossMajor
+        self._elementsCountAcrossMinor = elementsCountAcrossMinor
+        self._elementsCountTransition = elementsCountTransition
+        if self._isCore and self._elementsCountTransition > 1:
+            self._elementsCountAround = (elementsCountAround - 8 * (self._elementsCountTransition - 1))
```

Comment:
The reason why I had this in the tubenetworkmesh instead of the scaffolds' checkOptions is because I couldn't figure out a nice way of switching between one transition element case and multiple transition elements case.

For example, if I had an initial elementsCountAround of 16 in the UI, and I change elementsCountTransition to 2, then the new elementsCountAround would be 8. If this change was made in the UI, then  the values that get sent to tubenetworkmesh would be 8 and 2. Now, if I change the elementsCountTransition back to 1, then the values that get sent to tubenetworkmesh would be 8 and 1, which is different to the initial elementsCountAround of 16.

Any suggestions?

Reply:
Just need a priority e.g. number around > transition > across major.
I think it's best to make the number around guaranteed, then work with the transition ensuring its possible for the number around or reducing to 1, then fix the across-major number to work in with the others.